html/mycakeapp/src/Controller/AuctionController.php

```diff
@@ -187,10 +195,92 @@ public function home2()
+		}
+		// 落札者・出品者・ログインユーザーのidを変数に入れる
+		$bidder_id = $bidinfo['user_id'];
+		$seller_id = $biditem['user_id'];
+		$login_userid = $this->Auth->user('id');
+		// 落札者からのpost送信があった場合
+		if ($bidder_id === $login_userid && null === $shipping_to) {
+			$shipping = $this->Shipping->newEntity();
+			// ship.ctpから発送先の情報が送信されていれば、その発送先の情報をインスタンスに代入する
+			if ($this->request->is('post')) {
+				$shipping = $this->Shipping->patchEntity($shipping, $this->request->getData());
+				// Shippingを保存
+				if ($this->Shipping->save($shipping)) {
+					return $this->redirect(['action' => 'ship', $item_id]);
+				} else {
+					$this->Flash->error(__('保存に失敗しました。もう一度入力下さい。'));
+				}
+			}
+		} elseif($bidder_id === $login_userid && $shipping_to['is_shipped'] === true) {
+			return $this->redirect(['action' => 'receive', $item_id]);
+			// 出品者からのpost送信があった場合
+		} elseif ($seller_id === $login_userid && isset($shipping_to)) {
+			if ($this->request->is('post')) {
+				$shipping = $this->Shipping->find()->where(['Shipping.item_id' => $item_id])->first();
+				$shipping = $this->Shipping->patchEntity($shipping, ['is_shipped' => 1]);
+				// Shippingを更新
+				if ($this->Shipping->save($shipping)) {
+					return $this->redirect(['action' => 'ship', $item_id]);
+				} else {
+					$this->Flash->error(__('保存に失敗しました。もう一度入力下さい。'));
+				}
+			}
+		} else {
+			return $this->redirect(['action' => 'index']);
+		}
+		// viewに値を渡せるように保存。 同じ値が取れるものは排除する必要あり？e.g. bidinfo['biditem_id']でitem_idは取れる。
+			$this->set(compact('bidinfo', 'item_id', 'biditem', 'bidder_id', 'seller_id', 'login_userid', 'shipping_to',  'shipping'));
+		
+	}
+
+	public function receive($item_id = null)
+	{
+		$bidinfo = $this->Bidinfo->find()->where(['Bidinfo.biditem_id' => $item_id])->enableHydration(false)->toArray()[0];
+		$biditem = $this->Biditems->find()->where(['Biditems.id' => $item_id])->enableHydration(false)->toArray()[0];
+		$shipping_to = null;
+		$shipping = null;
+		if([] !== $this->Shipping->find()->where(['Shipping.item_id' => $item_id])->enableHydration(false)->toArray()){
+			$shipping_to = $this->Shipping->find()->where(['Shipping.item_id' => $item_id])->enableHydration(false)->toArray()[0];
+		}
+		// 落札者・ログインユーザーのidを変数に入れる
+		$bidder_id = $bidinfo['user_id'];
+		$login_userid = $this->Auth->user('id');
+		if ($bidder_id === $login_userid && $shipping_to['is_shipped'] === true) {
+			if ($this->request->is('post')) {
+				$shipping = $this->Shipping->find()->where(['Shipping.item_id' => $item_id])->first();
+				$shipping = $this->Shipping->patchEntity($shipping, ['is_received' => 1]);
+				// Shippingを更新
+				if ($this->Shipping->save($shipping)) {
+					return $this->redirect(['action' => 'receive', $item_id]);
+				} else {
+					$this->Flash->error(__('保存に失敗しました。もう一度入力下さい。'));
+				}
+			}
+		} else {
+			return $this->redirect(['action' => 'index']);
+		}
+		// viewに値を渡せるように保存。 同じ値が取れるものは排除する必要あり？e.g. bidinfo['biditem_id']でitem_idは取れる。
```

Comment:
info
基本、学習用のコメント等は消しておいてほうが良いです。
ただ、残しておきたいならこういった記述しておくのもありです。

```suggestion
		// @memo: viewに値を渡せるように保存。 同じ値が取れるものは排除する必要あり？e.g. bidinfo['biditem_id']でitem_idは取れる。
```

Reply:
こういったメモの残し方もあるのですね。
ご教示ありがとうございます！

Reply:
11cfba5
こちらのコミットで対応しました。